src/dft/backends/mklcpu/commit.cpp

```diff
@@ -31,109 +31,133 @@
-        }
+    commit_derived_impl(sycl::queue queue, dft::detail::dft_values<prec, dom> config_values)
+            : dft::detail::commit_impl(queue, backend::mklcpu),
+              status(DFTI_NOTSET) {
+        sycl::buffer<DFT_ERROR, 1> status_buffer{ &status, sycl::range<1>{ 1 } };
+
+        queue
+            .submit([&](sycl::handler& cgh) {
+                auto handle_obj =
+                    handle_buffer.template get_access<sycl::access::mode::read_write>(cgh);
+                auto status_obj =
+                    status_buffer.template get_access<sycl::access::mode::read_write>(cgh);
+
+                host_task<detail::kernel_name<mklcpu_desc_t>>(cgh, [=]() {
+                    std::int64_t local_err = DFTI_MEMORY_ERROR;
+                    if (config_values.dimensions.size() == 1)
+                        local_err =
+                            DftiCreateDescriptor(handle_obj.get_pointer(), mklcpu_prec, mklcpu_dom,
+                                                 1, config_values.dimensions[0]);
+                    else
+                        local_err = DftiCreateDescriptor(
+                            handle_obj.get_pointer(), mklcpu_prec, mklcpu_dom,
+                            config_values.dimensions.size(), config_values.dimensions.data());
+
+                    set_value(*handle_obj.get_pointer(), config_values);
+                    if (local_err != DFTI_NO_ERROR) {
+                        throw oneapi::mkl::exception("dft", "commit",
+                                                     "DftiCreateDescriptor failed");
+                    }
+
+                    local_err = DftiCommitDescriptor(*handle_obj.get_pointer());
+                    (*status_obj.get_pointer()) = local_err;
+                });
+            })
+            .wait();
+
+        status = status_buffer.template get_access<sycl::access::mode::read>()[0];
         if (status != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
-                                         "DftiCreateDescriptor failed");
+            throw oneapi::mkl::exception("dft", "commit", "DftiCommitDescriptor failed");
         }
 
-        set_value(handle, config_values);
-
-        status = DftiCommitDescriptor(handle);
-        if (status != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
-                                         "DftiCommitDescriptor failed");
-        }
+        device_handle = handle_buffer.template get_access<sycl::access::mode::read>()[0];
     }
 
-    virtual void* get_handle() noexcept override {
-        return handle;
+    void* get_handle() noexcept override {
+        return reinterpret_cast<void*>(device_handle);
     }
 
     virtual ~commit_derived_impl() override {
-        DftiFreeDescriptor((DFTI_DESCRIPTOR_HANDLE*)&handle);
+        DftiFreeDescriptor((mklcpu_desc_t*)&device_handle);
```

Comment:
```suggestion
        DftiFreeDescriptor(static_cast<mklcpu_desc_t*>(&device_handle));
```
but is the cast needed at all?